resolwe/flow/models.py

```diff
@@ -271,24 +270,20 @@ class Meta(BaseModel.Meta):
 
     """
 
+    requirements = JSONField(default=dict)
 
-def render_template(template_string, context):
-    """Render template based on Dango template language."""
-    template_headers = [
-        '{% load resource_filters %}',
-        '{% load process_fields %}',
-        '{% load mathfilters %}',
-    ]
 
-    custom_template_tags = getattr(settings, 'RESOLWE_CUSTOM_TEMPLATE_TAGS', [])
-    if not isinstance(custom_template_tags, list):
-        raise KeyError("`RESOLWE_CUSTOM_TEMPLATE_TAGS` setting must be a list.")
+def render_template(process, template_string, context):
+    """Render template using the specified expression engine."""
+    from .managers import manager
```

Comment:
For future reference:

Everything in the call to [`celery_run()`](https://github.com/genialis/resolwe/blob/master/resolwe/flow/tasks.py#L14) must be picklable/serializable.

For `data_id`, `script` and `verbosity`, this is not an issue, since they are `int`, `str` and `str` (respectively). The `executor` argument is of type [`BaseFlowExecutor`](https://github.com/genialis/resolwe/blob/master/resolwe/flow/executors/__init__.py#L65) which is a more complex type and we must ensure that it remains serializable. Note that it is pickled "by reference" which means that Celery workers must run the same `BaseFlowExecutor` code as the manager process.